src/vs/workbench/services/textMate/electron-browser/TMSyntax.ts

```diff
@@ -448,13 +463,13 @@ class TMTokenization implements ITokenizationSupport {
 			throw new Error('Unexpected: offsetDelta should be 0.');
 		}
 
-		// Do not attempt to tokenize if a line has over 20k
-		if (line.length >= 20000) {
+		// Do not attempt to tokenize if a line is too long
+		if (line.length >= this._longLineLength) {
 			if (!this._tokenizationWarningAlreadyShown) {
 				this._tokenizationWarningAlreadyShown = true;
-				this.notificationService.warn(nls.localize('too many characters', "Tokenization is skipped for lines longer than 20k characters for performance reasons."));
+				this.notificationService.warn(nls.localize('too many characters', 'Tokenization is skipped for long lines for performance reasons.'));
```

Comment:
Can you add a bit about how this can be configured now? "The length of a long line can be set by configuring editor.longLineLength in settings."